BlueprintUI/Sources/Animation/Transition.swift

```diff
@@ -0,0 +1,84 @@
+import UIKit
+
+
+/**
+ Wraps a content element and adds transitions when the element appears, disappears, or changes layout.
+ */
+public struct Transition<Wrapped:Element> : Element {
+
+    /// The transition to use when the wrapped element is first presented on screen.
+    public var onAppear: TransitionAnimation?
+    
+    /// The transition to use when the wrapped element is removed from the screen.
+    public var onDisappear: TransitionAnimation?
+    
+    /// The layout transition to use when an already on-screen element is updated.
+    public var layoutTransition: LayoutTransition
+
+    /// The element which will be transitioned in, out, or transitioned during layout.
+    public var wrapped : Wrapped
+    
+    //
+    // MARK: Initialization
+    //
+    
+    /// Creates a new `Transition` with both `onAppear` and `onDisappear` set to the provided `animation`.
+    public init(
+        animation: TransitionAnimation,
+        layout: LayoutTransition = .specific(AnimationOptions()),
+        wrapping element: Wrapped
```

Comment:
Consider making this a block that returns an element instead of a param? Generally this is more SwiftUI like, and allows inline config, and makes the API visually nicer when reading code.